apps/service_providers/tests/test_assistant_runnable.py

```diff
@@ -211,10 +234,100 @@ def test_assistant_uploads_new_file(create_and_run, retrieve_run, list_messages,
     assert result.output == "ai response"
     assert chat.get_metadata(chat.MetadataKeys.OPENAI_THREAD_ID) == thread_id
     message = chat.messages.filter(message_type="human").first()
-    assert message.metadata == {
-        "file_search": ["openai-file-1", "openai-file-2"],
-        "code_interpreter": ["openai-file-1", "openai-file-2"],
-    }
+    assert "openai-file-1" in message.metadata["openai_file_ids"]
+    assert "openai-file-2" in message.metadata["openai_file_ids"]
+
+
+@pytest.mark.django_db()
+@patch("apps.assistants.sync.get_and_store_openai_file")
+@patch("apps.service_providers.llm_service.runnables.AssistantExperimentRunnable._get_response_with_retries")
+@patch("openai.resources.beta.threads.runs.Runs.retrieve")
+@patch("openai.resources.beta.Threads.create_and_run")
+@patch("openai.resources.beta.threads.messages.Messages.list")
+def test_assistant_reponse_with_annotations(
+    list_messages, create_and_run, retrieve_run, get_response_with_retries, get_and_store_openai_file
+):
+    """Test that attachments on AI messages are being saved (only those of type `file_path`)
+    OpenAI doesn't allow you to fetch the content of the file that you uploaded, but this isn't an issue, since we
+    already have that file as an attachment on the chat object
+    """
+    from langchain.agents.openai_assistant.base import OpenAIAssistantFinish
+
+    session = ExperimentSessionFactory()
+    created_file = FileFactory(external_id="openai-file-1")
+    get_and_store_openai_file.return_value = created_file
+
+    thread_id = "test_thread_id"
+    run = _create_run(ASSISTANT_ID, thread_id)
+    cited_file = FileFactory(external_id="openai-file-2", team=session.team, name="existing.txt")
+    annotations = [
+        FilePathAnnotation(
+            end_index=174,
+            file_path=FilePath(file_id="openai-file-1"),
+            start_index=134,
+            text="sandbox:/mnt/data/file.txt",
+            type="file_path",
+        ),
+        FileCitationAnnotation(
+            end_index=174,
+            file_citation=FileCitation(file_id=cited_file.external_id, quote=""),
+            start_index=134,
+            text="【6:0†source】",
```

Comment:
why the special unicode `【】`? Is that what the API returns?

Reply:
Correct, this is what is being returned